src/main/java/org/esgi/boissibook/features/book/infra/repository/BookEntity.java

```diff
@@ -0,0 +1,181 @@
+    public BookEntity setPublisher(String publisher) {
+        this.publisher = publisher;
+        return this;
+    }
+
+    public BookEntity setPublishedDate(String publishedDate) {
+        this.publishedDate = publishedDate;
+        return this;
+    }
+
+    public BookEntity setDescription(String description) {
+        this.description = description;
+        return this;
+    }
+
+    public BookEntity setIsbn13(String isbn13) {
+        this.isbn13 = isbn13;
+        return this;
+    }
+
+    public BookEntity setLanguage(String language) {
+        this.language = language;
+        return this;
+    }
+
+    public BookEntity setImgUrl(String imgUrl) {
+        this.imgUrl = imgUrl;
+        return this;
+    }
+
+    public BookEntity setPages(int pages) {
+        this.pages = pages;
+        return this;
+    }
+
+    public String id() {
+        return id;
+    }
+
+    public String apiId() {
+        return apiId;
+    }
+
+    public String title() {
+        return title;
+    }
+
+    public List<String> authors() {
+        return authors;
+    }
+
+    public String publisher() {
+        return publisher;
+    }
+
+    public String publishedDate() {
+        return publishedDate;
+    }
+
+    public String description() {
+        return description;
+    }
+
+    public String isbn13() {
+        return isbn13;
+    }
+
+    public String language() {
+        return language;
+    }
+
+    public String imgUrl() {
+        return imgUrl;
+    }
+
+    public int pages() {
+        return pages;
+    }
+
+    @Override
+    public String toString() {
+        return "BookEntity{" +
+            "id='" + id + '\'' +
+            ", apiId='" + apiId + '\'' +
+            ", title='" + title + '\'' +
+            ", authors=" + authors +
+            ", publisher='" + publisher + '\'' +
+            ", publishedDate='" + publishedDate + '\'' +
+            ", description='" + description + '\'' +
+            ", isbn13='" + isbn13 + '\'' +
+            ", language='" + language + '\'' +
+            ", imgUrl='" + imgUrl + '\'' +
+            ", pages=" + pages +
+            '}';
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        BookEntity that = (BookEntity) o;
+        return pages == that.pages && Objects.equals(id, that.id) && Objects.equals(apiId, that.apiId) && Objects.equals(title, that.title) && Objects.equals(authors, that.authors) && Objects.equals(publisher, that.publisher) && Objects.equals(publishedDate, that.publishedDate) && Objects.equals(description, that.description) && Objects.equals(isbn13, that.isbn13) && Objects.equals(language, that.language) && Objects.equals(imgUrl, that.imgUrl);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, apiId, title, authors, publisher, publishedDate, description, isbn13, language, imgUrl, pages);
+    }

```

Comment:
Pour moi les entités BDD devrait pas override equal et hashcode. Je suis d'accord que l'identifiant c'est l'ID. mais alors c'est que l'ID. notre entity elle est dans le domain ici je le vois comme un object classique

Reply:
d'accord avec Swann je sais que nous on l'a pas fait mais on pourrait créé des VO pour les id mais du coup laisser des objets "entity" (je parle pas de DB) mais ça https://enterprisecraftsmanship.com/posts/entity-vs-value-object-the-ultimate-list-of-differences/#:~:text=The%20main%20difference%20between%20entities,while%20value%20objects%20don't.
pour les objets du domaine.

Reply:
c'était une erreur pour l'entité, par contre pour le domain je dois faire une vérification que sur l'id du coup ?